### PR TITLE
support krb5 for hdfs

### PR DESCRIPTION
**Pull request by white-cc**
### Changelog category :
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

add krb5 support for hdfs

### Documentation entry for user-facing changes

     add conf in server.xml in <yandex></yandex>

    <hadoop_kerberos_keytab>/path/to/keytab</hadoop_kerberos_keytab>
    <hadoop_kerberos_principal>kerberos_principal</hadoop_kerberos_principal>
    <hadoop_security_kerberos_ticket_cache_path>/ticket/cache/path</hadoop_security_kerberos_ticket_cache_path>

    add conf in hdfs3.xml like 
    <property>
        <name>hadoop.security.authentication</name>
        <value>KERBEROS</value>
    </property>
    <property>
        <name>dfs.use.internal.credential.format</name>
        <value>false</value>
    </property>
    <property>
        <name>dfs.data.transfer.protection</name>
        <value>authentication</value>
    </property>
    <property>
        <name>dfs.block.access.token.enable</name>
        <value>true</value>
    </property>

### CODE REVIEW

**Comment by WillemJiang** [comment]
@white-cc Please add the license header [as this](https://github.com/ByConity/ByConity/blob/master/src/DaemonManager/DaemonJobServerBGThread.cpp#L1)  on the files you created.

File: src/Storages/HDFS/HDFSAuth.cpp [hunk 1]
```diff
@@ -1,3 +1,18 @@
+/*
+ * Copyright (2022) Bytedance Ltd. and/or its affiliates
```
Comment (WillemJiang) [inline src/Storages/HDFS/HDFSAuth.cpp:2]: Change it to “Copyright 2023 Bytedance Ltd. and/or its affiliates” ， as it is added this year.

File: src/Access/KerberosInit.cpp [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/*
```
Comment (WillemJiang) [inline src/Access/KerberosInit.cpp:17]: We can remove the below lines if we add the  Bytedance Copyright below the ClickHouse one.

File: src/Access/KerberosInit.cpp [hunk 3]
```diff
@@ -1,3 +1,24 @@
...
+ * Copyright 2016-2023 ClickHouse, Inc.
```
Comment (WillemJiang) [inline src/Access/KerberosInit.cpp:2]: You can put the ‘Copyright 2023 Bytedance Ltd. and/or its affiliates.’ below  the  copyright of ClickHouse.

**Review by WillemJiang — CHANGES_REQUESTED** [review]

**Review by hustnn — APPROVED** [review]

**Pull request merged**